dynomite/restore.go

```diff
@@ -0,0 +1,93 @@
+package dynomite
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"time"
+
+	"github.com/majewsky/schwift"
+	"github.com/sapcc/go-bits/logg"
+)
+
+// Restore downloads the latest backup to disk
+func Restore(containerName, prefix string) error {
+	currentTime := time.Now()
+
+	iter, err := newIterator(containerName, prefix)
+	if err != nil {
+		return err
+	}
+
+	// Get the last backup
+	objects, err := iter.NextPage(1)
+	if err != nil {
+		return err
+	}
+
+	if len(objects) == 0 {
+		return fmt.Errorf("Restore failed - No Backups found in: %s", containerName+prefix)
+	}
+
+	logg.Info("Downloading %s", objects[0].FullName())
+	data, err := objects[0].Download(nil).AsByteSlice()
+
+	err = ioutil.WriteFile("/data/dump.rdb", data, 0644)
```

Comment:
I don't know how large these dumps get. If it's relatively small, that's okay. If it gets rather big, you might want to avoid collecting everything into a large []byte first:

```go
data, err := objects[0].Download(nil).AsReadCloser()
file, err := os.Create("/data/dump.rdb")
err = io.Copy(file, data)
err = data.Close()
err = file.Close()
```

Error handling elided for brevity.

Reply:
Switched to Reader